From e984f59f8d8c2363d9ffa6682d5961e245164a3a Mon Sep 17 00:00:00 2001
From: Miaomiao Chen <miaochen@codeaurora.org>
Date: Fri, 2 Aug 2019 10:14:01 +0800
Subject: [PATCH 16/42] omxvideodec: resolve video black screen

Add error handle to cover other case of port setting change event,
which will send error message to gst-omx, makes the componet state to
invalid by accident.

Change-Id: I9eeba92a7ee8af6ab57d826c73b34e0fa17c13bf
Signed-off-by: Miaomiao Chen <miaochen@codeaurora.org>
---
 omx/gstomx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 907eba3..bd7110d 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -775,6 +775,10 @@ EventHandler (OMX_HANDLETYPE hComponent, OMX_PTR pAppData, OMX_EVENTTYPE eEvent,
         msg->content.port_settings_changed.port = index;
         GST_DEBUG_OBJECT (comp->parent, "%s rectangle changed (port index: %u)",
           comp->name, (guint) msg->content.port_settings_changed.port);
+      } else {
+        GST_WARNING_OBJECT (comp->parent, "Never handle event:0x%x param:%d", msg->type, nData2);
+        g_slice_free (GstOMXMessage, msg);
+        break;
       }
 
       gst_omx_component_send_message (comp, msg);
-- 
2.7.4

